packages/fxa-settings/src/pages/ResetPasswordRedesign/CompleteResetPasswordOtp/container.tsx

```diff
@@ -0,0 +1,207 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { RouteComponentProps, useLocation, useNavigate } from '@reach/router';
+
+import { useValidatedQueryParams } from '../../../lib/hooks/useValidate';
+import {
+  Integration,
+  isSyncDesktopV3Integration,
+  isSyncOAuthIntegration,
+  useAccount,
+  useConfig,
+  useFtlMsgResolver,
+} from '../../../models';
+import { KeyStretchExperiment } from '../../../models/experiments';
+
+import CompleteResetPasswordOtp from '.';
+import {
+  AccountResetData,
+  CompleteResetPasswordLocationState,
+} from './interfaces';
+import GleanMetrics from '../../../lib/glean';
+import firefox from '../../../lib/channels/firefox';
+import { getLocalizedErrorMessage } from '../../../lib/auth-errors/auth-errors';
+import { useState } from 'react';
+
+// This component is used for both /complete_reset_password and /account_recovery_reset_password routes
+// for easier maintenance
+
+const CompleteResetPasswordOtpContainer = ({
+  integration,
+}: {
+  integration: Integration;
+} & RouteComponentProps) => {
+  const keyStretchExperiment = useValidatedQueryParams(KeyStretchExperiment);
+
+  const account = useAccount();
+  const config = useConfig();
+  const ftlMsgResolver = useFtlMsgResolver();
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const searchParams = location.search;
+
+  if (!location.state) {
+    navigate(`/reset_password${searchParams}`);
+  }
+
+  const {
+    code,
+    email,
+    token,
+    accountResetToken,
+    emailToHashWith,
+    kB,
+    recoveryKeyId,
+  } = location.state as CompleteResetPasswordLocationState;
+
+  const hasConfirmedRecoveryKey = !!(
+    accountResetToken &&
+    email &&
+    kB &&
+    recoveryKeyId
+  );
+
+  const isResetWithoutRecoveryKey = !!(code && token);
+
+  // TODO in FXA-7890 handle case where hasAccountRecoveryKey could not be determined
+  // and show banner to use account recovery key
+  // add metrics to track if users ever click on the link
+
+  const handleNavigationWithRecoveryKey = () => {
+    navigate(`/reset_password_with_recovery_key_verified${location.search}`, {
+      replace: true,
```

Comment:
Hm, I see where we're using `replace: true` in other places but we should only use it when we don't want the user to be able to navigate back to this page manually. Just checking that's the case here, maybe it is, but maybe for `handleNavigationWithoutRecoveryKey` we want them to be able to click back if they want to?